### PR TITLE
#159987715 Tag articles on creation

### PR DESCRIPTION
**Pull request by emekafredy**
# What does this PR do?
Create endpoints for users profile

# Description of Task to be completed?
### Create endpoint
- Create tags ``` POST api/articles/tags ```
### Modify endpoint
- Article creation with tags ``` POST api/articles ```

# How should this be manually tested?
- Clone the repo
- cd into the 'Thor-ah'
- Run 'npm install'
- Use the .env.sample format to create a .env file
- Run 'npm run migrate'
- Run 'npm run start:dev'

# Any background context you want to provide?
N/A

# What are the relevant pivotal tracker stories?
[159987715](https://www.pivotaltracker.com/story/show/159987715)

# Screenshots (if appropriate)
N/A

# Questions:
N/A

### CODE REVIEW

File: server/tests/controllers/articles.js [hunk 1]
```diff
@@ -179,7 +258,7 @@ describe('Articles controller', () => {
         });
     });
 
-    it('should return error if slug not found', (done) => {
+    it('shouuld return error if slug not found', (done) => {
```
Comment (CEOehis) [inline server/tests/controllers/articles.js]: @emekafredy typo here and on line `237`

File: server/tests/controllers/articles.js [hunk 2]
```diff
@@ -63,18 +108,52 @@ describe('Articles controller', () => {
...
+        .set('Content-Type', 'application/json')
+        .send(Article2)
+        .end((err, res) => {
+          res.should.have.status(200);
+          const { newArticleAlert } = res.body;
+          newArticleAlert.should.be.a('object');
+          newArticleAlert.createdArticle.should.be.an('object');
+          newArticleAlert.createdArticle.should.have.property('slug');
+          newArticleAlert.createdArticle.should.have.property('title');
+          newArticleAlert.createdArticle.should.have.property('description');
```
Comment (CEOehis) [inline server/tests/controllers/articles.js]: @emekafredy I think you can make this code more consise by using destructuring to extract `createdArticle` from `newArticleAlert` as well. lines `114 - 121` and `139 - 142`

**Review by CEOehis — CHANGES_REQUESTED** [review]

File: server/tests/controllers/articles.js (hunk 1 above)
Comment (emekafredy) [reply server/tests/controllers/articles.js]: Okay. Thanks. I will fix this

File: .hound.yml [hunk 3]
```diff
@@ -4,5 +4,5 @@ scss:
 
 eslint:
   enabled: true
-  config_file: .eslintrc
+  config_file: .eslintrc.json
```
Comment (lauragift21) [inline .hound.yml]: @emekafredy You can revert the changes to this file. So it doesn't cause conflicts with what we have in the develop branch.

Comment (emekafredy) [reply .hound.yml]: Okay, thanks. I'm on it

File: server/tests/controllers/articles.js (hunk 2 above)
Comment (emekafredy) [reply server/tests/controllers/articles.js]: Okay. I'm also working on this
